op-program/host/prefetcher/prefetcher_test.go

```diff
@@ -569,7 +569,7 @@ func TestRetryWhenNotAvailableAfterPrefetching(t *testing.T) {
 	_, l1Source, l1BlobSource, l2Cl, kv := createPrefetcher(t)
 	putsToIgnore := 2
 	kv = &unreliableKvStore{KV: kv, putsToIgnore: putsToIgnore}
-	prefetcher := NewPrefetcher(testlog.Logger(t, log.LevelInfo), l1Source, l1BlobSource, l2Cl, kv)
+	prefetcher := NewPrefetcher(testlog.Logger(t, log.LevelInfo), l1Source, l1BlobSource, l2Cl, kv, false)
```

Comment:
We should add some tests that enable the new bulk fetching methods.

Reply:
In particular, could we add a test that shows the fetching does actually fallback to debug_dbGet if the experimental request either returns an error or fails to populate the required preimage.  Looking at the code I'm a bit suspicious that it will actually only try the experimental request and never fallback but I could be wrong.

Reply:
Added a test here: [c9a6fa5](https://github.com/ethereum-optimism/optimism/pull/12559/commits/c9a6fa51f58f8f4ea7bdd6a4475bd9993215e310)